### ISSUE TITLE
enter key submits instead of breaking the line

### ISSUE THREAD

**Issue by ygrek** [8VIM/8VIM#568]
In some apps (e.g. slack) pressing enter (the bottom quadrant) sends the message, instead of introducing the line break. Other keyboard (builtin, FUTO, etc) don't have such behaviour. It is not universal accross apps (e.g. in Signal it does introduce line break as expected), so maybe it is slack shenanigans, but other keyboards somehow are not affected, so it puts 8vim at disadvantage (i have to switch to other keyboard to compose anything longer than one line). Ctrl and Fn modifier are not the reason (toggled them on and off in all combinations).

 - OS: android
 - Browser: no browser, official slack app
 - Device: Pixel 7

**Comment by GitAlasGo** [comment]
Same in unexpected keyboard: https://github.com/Julow/Unexpected-Keyboard/issues/822

**Comment by giblet75** [comment]
I also find this is a little bit of a pain when doing corrections (e.g. I have a typo on a word, so I move back to a letter in the middle of the word, delete the letter and then type a new one. Suddenly my word is split into two and I need to delete the space.

Suggestion - I don't know if its feasible to add a click after the lift to negate the space, or an alternative ending position that can be used to denote no space...I do like the idea of dragging out to the enter key to send enter - that's what I expected to happen when I first used this keyboard.

Note: I do think this is an excellent app, and it's been fun learning how to use it.

**Comment by janIpen** [comment]
> I also find this is a little bit of a pain when doing corrections (e.g. I have a typo on a word, so I move back to a letter in the middle of the word, delete the letter and then type a new one. Suddenly my word is split into two and I need to delete the space.
> 
> Suggestion - I don't know if its feasible to add a click after the lift to negate the space, or an alternative ending position that can be used to denote no space...I do like the idea of dragging out to the enter key to send enter - that's what I expected to happen when I first used this keyboard.
> 
> Note: I do think this is an excellent app, and it's been fun learning how to use it.

if you *don't* end your touch in the centre circle, no space is added. 👍

**Comment by fedde-s** [comment]
@janIpen True, I use that all the time, works great once you get used to it! The point about the enter key making some apps behave more zealously than they would with other keyboards' enter keys still stands though. I have to switch to a different keyboard just to type a multi-line message, which I fear could put other users (and potential code/doc/idea contributors) off continuing to use the app.

**Comment by janIpen** [comment]
> @janIpen True, I use that all the time, works great once you get used to it! The point about the enter key making some apps behave more zealously than they would with other keyboards' enter keys still stands though. I have to switch to a different keyboard just to type a multi-line message, which I fear could put other users (and potential code/doc/idea contributors) off continuing to use the app.

yeah, I made an issue about that before I saw this one. I think the project might be dead though...

**Comment by fedde-s** [comment]
> I think the project might be dead though...

Less than three months ago someone did take over to perform some maintenance tasks, some pull request replies and a feature I'd personally turn off if it gets released. They might just have been busy since then 🙂  Meanwhile there's several people engaging with the project on GitHub, so although I myself currently have no direct Android keyboard app development experience nor the ability to promise that I can be reliably available every so often, I think there's hope that, even in the worst case, someone could take over again to keep this cool thing maintained enough to be usable.

**Comment by janIpen** [comment]
okay yay
